ntbs-service-unit-tests/DataMigration/ImportValidatorContactTracingTest.cs

```diff
@@ -0,0 +1,505 @@
+                ValidationMessages.ContactTracingAdultsLatentTB,
+                ValidationMessages.ContactTracingAdultsStartedTreatment,
+                ValidationMessages.ContactTracingAdultsFinishedTreatment
+            });
+
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.AdultsIdentified), valueEntered);
+
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.AdultsScreened), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.AdultsActiveTB), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.AdultsLatentTB), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.AdultsStartedTreatment), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.AdultsFinishedTreatment), 3);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(1)]
+        public async Task ImportValidatorCleansContactTracingValuesBasedOn_ChildrenIdentified(int? valueEntered)
+        {
+            // Arrange
+            var notification = CreateBasicNotification();
+
+            notification.ContactTracing = new ContactTracing
+            {
+                ChildrenIdentified = valueEntered,
+                ChildrenScreened = 3,
+                ChildrenActiveTB = 3,
+                ChildrenLatentTB = 3,
+                ChildrenStartedTreatment = 3,
+                ChildrenFinishedTreatment = 3,
+            };
+
+            // Act
+            var validationResults = await _importValidator.CleanAndValidateNotification(null, RunId, notification);
+
+            // Assert
+            // Validation errors
+            var errorMessages = validationResults.Select(r => r.ErrorMessage).ToList();
+            Assert.Empty(errorMessages);
+
+            AssertWarningsGiven(notification, new[]
+            {
+                ValidationMessages.ContactTracingChildrenScreened,
+                ValidationMessages.ContactTracingChildrenActiveTB,
+                ValidationMessages.ContactTracingChildrenLatentTB,
+                ValidationMessages.ContactTracingChildrenStartedTreatment,
+                ValidationMessages.ContactTracingChildrenFinishedTreatment
+            });
+
+            AssertContactTracingValuePreserved(notification, nameof(ContactTracing.ChildrenIdentified), valueEntered);
+
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.ChildrenScreened), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.ChildrenActiveTB), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.ChildrenLatentTB), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.ChildrenStartedTreatment), 3);
+            AssertContactTracingValueRemoved(notification, nameof(ContactTracing.ChildrenFinishedTreatment), 3);
```

Comment:
This has now become complicated enough, that I do think it would be worth pulling it out into its own method

Reply:
I've had a go at doing that, what do you think?